src/equations/equations.jl

```diff
@@ -110,6 +110,7 @@ struct BoundaryConditionDirichlet{B}
   boundary_value_function::B
 end
 
+# Dirichlet-type boundary condition for use with TreeMesh or CurvedMesh
 @inline function (boundary_condition::BoundaryConditionDirichlet)(u_inner, orientation, direction,
```

Comment:
I think you should restrict the argument `orientation::Integer` to make the purpose clear.

Reply:
In the curved implementation this variable can be orientation or the normal direction so I changed the variable name to `orientation_or_normal`